### PR TITLE
Add pre-commit-hooks

### PR DESCRIPTION
**Pull request by Bouni**
This is meant to be a starting point. We'll add the hooks one after another to prevent to many changes to be made at a time.

Here we start with `black` as  the first pre-commit-hook.

make sure you have installed pre-commit (` pip install pre-commit`) then simply run `pre-commit install` and you set.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+repos:
```
Comment (kbabioch) [inline .pre-commit-config.yaml:1]: We should add `---` at the beginning of this document in order to address the issue:

1:1 [document-start] missing document start "---"

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  - repo: https://github.com/psf/black
+    rev: 22.12.0
+    hooks:
+      - id: black
+        files: ^((luxtronik|pylint|scripts|tests)/.+)?[^/]+\.py$
+  # - repo: https://github.com/charliermarsh/ruff-pre-commit
```
Comment (kbabioch) [inline .pre-commit-config.yaml:7]: > 7:3 [comments-indentation] comment not indented like content

**Review by kbabioch — APPROVED** [review]

**Comment by kbabioch** [comment]
One more thing: This file is ending in `.yaml`, other YAML files in our repository end in `.yml`. Should we try to stay consistent here?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+---
```
Comment (kbabioch) [inline .pre-commit-config.yaml:1]: One more thing: This file is ending in `.yaml`, other YAML files in our repository end in `.yml`. Should we try to stay consistent here?

Comment (Bouni) [reply .pre-commit-config.yaml:1]: I just tried and pre-commit expects a .yaml file, .yml is not recognized 🤷🏽‍♂️

**Pull request merged**